sqlmodel/main.py

```diff
@@ -141,19 +204,78 @@ def Field(
     max_length: Optional[int] = None,
     allow_mutation: bool = True,
     regex: Optional[str] = None,
+    # The following are specific to SQLAlchemy:
+    name: Optional[str] = None,
+    type_: Union[TypeEngine, UndefinedType] = Undefined,  # type: ignore[type-arg]
+    autoincrement: Union[bool, str] = "auto",
+    doc: Optional[str] = None,
+    key: Union[str, UndefinedType] = Undefined,  # `Column` default is `name`
+    index: Optional[bool] = None,
+    info: Union[Dict[str, Any], UndefinedType] = Undefined,  # `Column` default is `{}`
+    nullable: Union[
+        bool, UndefinedType
+    ] = Undefined,  # `Column` default depends on `primary_key`
+    onupdate: Any = None,
     primary_key: bool = False,
-    foreign_key: Optional[Any] = None,
-    unique: bool = False,
-    nullable: Union[bool, UndefinedType] = Undefined,
-    index: Union[bool, UndefinedType] = Undefined,
-    sa_column: Union[Column, UndefinedType] = Undefined,  # type: ignore
-    sa_column_args: Union[Sequence[Any], UndefinedType] = Undefined,
-    sa_column_kwargs: Union[Mapping[str, Any], UndefinedType] = Undefined,
+    server_default: Union[FetchedValue, str, TextClause, None] = None,
+    server_onupdate: Optional[FetchedValue] = None,
+    quote: Union[
+        bool, None, UndefinedType
+    ] = Undefined,  # `Column` default not (fully) defined
+    unique: Optional[bool] = None,
+    system: bool = False,
+    comment: Optional[str] = None,
+    foreign_key: Optional[str] = None,
```

Comment:
```suggestion
    foreign_key: Union[str, ForeignKey, None] = None,
```

Reply:
Hello, I think your PR is very good, and I really hope to be merged into the main branch as soon as possible.

Reply:
Thank you @amisadmin and thank you for the suggested additions.

I understand, what you are trying to do. You want to be able to not just specify the foreign key via a string like this:
```python
Field(foreign_key="table.key", index=True)
```
But also explicitly via an SQLAlchemy `ForeignKey` instance, so that you can specify additional parameters for the foreign key like this for example:
```python
Field(foreign_key=ForeignKey("table.key", ondelete="CASCADE"), index=True)
```

However, this is not really in line with my intent here. The purpose of my PR is to allow you to basically treat an SQLModel `Field` almost the same as an SQLAlchemy `Column`. With my proposed changes, you _should_ therefore be able to just do this:
```python
Field(ForeignKey("table.key", ondelete="CASCADE"), index=True)
```
Which is exactly how you would do it with a `Column` in pure SQLAlchemy.

My idea was that the `foreign_key` parameter should remain as a "shortcut" to allow you to simply specify the referenced table and key as a string, if you don't need further customization for the foreign key. (I assume this was @tiangolo's original idea as well.)

The problem is that your proposed additions basically introduce a **3rd way** to accomplish the same thing. And this goes hardcore against the Zen of Python...

> There should be one-- and preferably only one --obvious way to do it.

And now that I think about it, I am not even sure, if keeping the `foreign_key` parameter **at all** is such a good idea. I suppose this is debatable.

In any case, although I appreciate your idea/effort, it doesn't fit with my proposition. I hope this makes sense.

If you want to continue pursuing this route, you may want to wait, until @tiangolo decides, whether this PR should be merged at all, and then propose your own additions in a separate PR.

Reply:
Thank you for your reply and I agree with you. I think your example is a good solution to my needs.